app/controllers/katello/api/registry/registry_proxies_controller.rb

```diff
@@ -85,6 +84,223 @@ def registry_authorize
+      end
+      product = @organization.products.where("LOWER(label) = '#{prod_label}'") # convert to lowercase
+      # reject ambiguous products (possible due to lowercase conversion)
+      if product.length > 1
+        return render_podman_error(
+          "NAME_INVALID",
+          "Product label '#{prod_label}' is ambiguous. Try using an id-based container name.",
+          :conflict
+        )
+      end
+      if product.length == 0
+        return render_podman_error(
+          "NAME_UNKNOWN",
+          "Product not found: '#{prod_label}'",
+          :not_found
+        )
+      end
+      @product = product.first
+      true
+    end
+
+    def check_blob_push_product_id(props)
+      prod_id = props[:product]
+      unless prod_id.present? && prod_id == prod_id.to_i.to_s
+        return render_podman_error(
+          "NAME_INVALID",
+          "Invalid format. Product id must be an integer without leading zeros.",
+          :bad_request
+        )
+      end
+      @product = @organization.products.find(prod_id.to_i)
+      if @product.nil?
+        return render_podman_error(
+          "NAME_UNKNOWN",
+          "Product id not found: '#{prod_id}'",
+          :not_found
+        )
+      end
+      true
+    end
+
+    def check_blob_push_container(props)
+      # All incorrect params should have been rejected by this point, assume ok
+      @container_name = props[:name]
+      @container_path_input = "#{props[:organization]}/#{props[:product]}/#{props[:name]}"
+      @container_path_actual = "#{@organization.label}/#{@product.label}/#{@container_name}"
+      return true
+    end
+
+    def create_container_repo_if_needed
+      if @product.root_repositories.where(label: @container_name).empty?
+        root = @product.add_repo(
+          name: @container_name,
+          label: @container_name,
+          download_policy: 'immediate',
+          content_type: Repository::DOCKER_TYPE,
+          unprotected: true,
+          is_container_push: true
+        )
+        sync_task(::Actions::Katello::Repository::CreateRoot, root, @container_path_input)
+      end
+    end
+
+    def handle_blob_push_cleanup
+      # if we're returning a successful code, index content and set version href using pulp api
+      if response.status.between(200, 299)
+        root_repo = @product.root_repositories.where(label: @container_name)&.first
+        instance_repo = root_repo&.library_instance
+
+        unless root_repo.present? && instance_repo.present?
+          return render_podman_error(
+            "BLOB_UPLOAD_UNKNOWN",
+            "Could not locate local uploaded repository for content indexing.",
+            :not_found
+          )
+        end
+
+        api = ::Katello::Pulp3::Repository.api(SmartProxy.pulp_primary, ::Katello::Repository::DOCKER_TYPE).container_push_api
+        latest_version_href = api.list(name: @container_path_input).results&.first&.latest_version_href
+
+        unless latest_version_href.blank?
```

Comment:
Should this be:

```suggestion
        if latest_version_href.blank?
```

Also one note: I like `empty?` because it covers nil, empty strings, and empty arrays.

Reply:
`if latest_version_href.empty?` thank you!